src/extensions/extension-discovery.ts

```diff
@@ -266,18 +266,30 @@ export class ExtensionDiscovery {
 
     logger.info(`${logModule} loading extensions from ${extensionInstaller.extensionPackagesRoot}`);
 
-    if (fs.existsSync(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"))) {
+    if (await fs.pathExists(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"))) {
       await fs.remove(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"));
     }
```

Comment:
Isn't this technically a race condition? Shouldn't it be something like the following:
```suggestion
    try {
      await fs.remove(path.join(extensionInstaller.extensionPackagesRoot, "package-lock.json"));
    } catch (error) {
      if (error.code !== "ENOENT") { // ignore file not found errors.
        throw error;
      }
    }
```

Reply:
fs.remove doesn't actually throw if the path if missing, so we could just call that without the if. I didn't want to change the code too much, but I guess it's fine to change it that way.